engine-precompiles/src/alt_bn256.rs

```diff
@@ -42,21 +42,144 @@ mod consts {
 
     /// Pair element length.
     pub(super) const PAIR_ELEMENT_LEN: usize = 192;
+
+    pub(super) const SCALAR_PART_LEN: usize = SCALAR_LEN / 2;
+
+    /// Size of BN scalars.
+    pub(super) const SCALAR_LEN: usize = 32;
+
+    /// Half the size of a point size.
+    pub(super) const POINT_PART_LEN: usize = POINT_LEN / 2;
+
+    /// Size of BN points.
+    pub(super) const POINT_LEN: usize = 64;
+
+    /// Size of BN pairs.
+    pub(super) const POINT_PAIR_LEN: usize = 128;
+
+    /// Output length.
+    pub(super) const OUTPUT_LEN: usize = 64;
+}
+
+#[cfg(feature = "contract")]
+trait SdkEncode {
```

Comment:
Maybe this should be named differently. Initially I was confused because we have other traits called `Sdk*` which are used to create output from the engine, so that's what I thought these would be for. Then I noticed that the implementations of this trait use little endian encoding instead of big endian, so I was concerned the implementation might be wrong. But then I got to where it was used and saw it is for passing into the NEAR host functions, not creating output, and NEAR does have a habit of preferring little endian.

Long story short, I think `HostFnEncode` might be a better name.

Reply:
Done in https://github.com/aurora-is-near/aurora-engine/pull/540/commits/55bdbe6d8b7060e8f373d42ce895ed17aabee3e3